_posts/2021-05-30-cloudnative-bot.md

```diff
@@ -0,0 +1,158 @@
+        controller.Invoke(topic, &data)
+      default:
+        log.Printf("Got something else %v", v)
+      }
+    }
+  }()
+```
+
+The complete source code can see here: https://github.com/cpanato/cloudnative-bot/blob/main/cmd/twitter-stream/main.go.
+
+To deploy this service, we are using `ko` with is an excellent tool, and we don't need to define a docker file; you can build and push the container right away and have your Kubernetes manifests configured with that image.
+
+One way to use `ko` is to define your deployment manifest, and in the image field, have the `ko` syntax, for example:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cloudnative-bot-stream
+  labels:
+    app.kubernetes.io/name: cloudnative-bot-stream
+    app.kubernetes.io/managed-by: honk
+spec:
+  replicas: 1
+    spec:
+      containers:
+      - name: stream
+        image: ko://github.com/cpanato/cloudnative-bot/cmd/twitter-stream
+        imagePullPolicy: IfNotPresent
+```
+Note: some lines were removed from the manifest to make it more clear.
+
+After defining the deployment, you can run `ko,` and the output will be the updated manifests ready to be deployed and the image pushed to the registry.
+
+```console
+$ export KO_DOCKER_REPO=YOUR_REGISTRY_NAME
+$ ko resolve -f  deployment.yaml > release.yaml
+```
+
+### Using OpenFaas functions for the slash commands
+
+We are using OpenFaas to deploy the slash command functions for the bot, and those will be triggered by the connector-sdk code we show above.
+
+The `CloudNativeBot` have currently three functions:
+
+	* `/cloudnativetv` - returns a random logo for the CNCF CloudNative TV shows
+	* `/k8s-patch-schedule` - returns the patch schedule for the active release branches fro Kubernetes
+	* `/kubecon-random-video`- returns a random KubeCon video
+
+The code for each function you can found in the CloudNativeBot GitHub Project: https://github.com/cpanato/cloudnative-bot
+
+An example of how that works:
+
+* A Twitter user post a message `/kubecon-random-video`
+* The stream listener will catch that and will trigger the function
+* The specific function, in this case, the `kubecon-random-video` will grab a random video from Youtube and post a reply message on Twitter.
+
+
+  ![https://twitter.com/comedordexis/status/1398286724569182219](/images/2021-05-cnbot/twitter-bot-example.png)
+
+
+### Wrap up
+
+Using OpenFaas and the connector-sdk we were able to bootstrap the project in a quick way and have each function very well defined and scoped.
+
+`ko` is a handy tool that helps to build and push images for golang projects without a need to define the Dockerfile.
+
+Using all those tools and infrastucture we can learn a bit how everything can work together and have insights to use in our projects and products.
+
+If you like it, post a message for the bot and watch a Kubecon video suggestion or discover the logos of the CNCF CloundNative TV shows!
```

Comment:
Let's then have a few bullets for actions to take next. Put Ko into that list instead of in your main conclusion.

Then also add a link to the blog post on Golang with OpenFaaS -> https://www.openfaas.com/blog/golang-serverless/

And perhaps a link to KubeCon since that's what your bot is about?

How about a link to your bot and asking people to follow it?